### PR TITLE
Replace twisted with aiohttp

### LINKED ISSUE #38: Replace `aiounittest` with `IsolatedAsyncioTestCase`
Python 3.8 (which is the minimal supported version for this project) natively supports async test cases: https://docs.python.org/3/library/unittest.html#unittest.IsolatedAsyncioTestCase

We should ditch aiounittest and replace it with the test case from `unittest`.

### PR DESCRIPTION
**Pull request by babolivier**
I'm also using this as an opportunity to switch async test cases to `unittest`'s native async test case (since that's available as of Python 3.8), which fixes #38.

See https://github.com/matrix-org/matrix-content-scanner-python/pull/37#issuecomment-1308933114 for context into what triggered this work.

### CODE REVIEW

**Review by squahtx — CHANGES_REQUESTED** [review]
Looks good.

File: matrix_content_scanner/servlets/thumbnail.py [hunk 1]
```diff
@@ -11,43 +11,38 @@
...
+    @web_handler
+    async def handle_thumbnail(
+        self, request: web.Request
+    ) -> Tuple[int, _BytesResponse]:
+        """Handles GET requests to .../thumbnail/serverName/mediaId"""
+        media_path = request.match_info["media_path"]
 
-        # request.args stores all keys and values as bytes. However, we want them to be
-        # strings going forward, so we convert them now.
+        # request.query is a multidict, however this isn't easily
```
[the remark this answers is not in the document]
Comment (babolivier) [reply matrix_content_scanner/servlets/thumbnail.py:35]: Ah, this block is actually some poc I forgot to remove, thanks for pointing out!

**Comment by babolivier** [comment]
I've also renamed the `servlets` module into `http` as "servlets" felt a bit antiquated a term to me and aiohttp uses "handlers" which is more in line with more recent web applications.

**Comment by babolivier** [comment]
> I've also renamed the `servlets` module into `http` as "servlets" felt a bit antiquated a term to me and aiohttp uses "handlers" which is more in line with more recent web applications. Happy to revert if people feel unhappy about this decision.

Nevermind, actually it probably makes things more confusing when comparing with other projects of ours, and also causes conflicts and issues, so I've reverted that change.

**Review by squahtx — APPROVED** [review]

**Pull request merged**